### ISSUE TITLE
[Bug]: ERR_NO_BUFFER_SPACE in browser killing entire connection

### ISSUE THREAD

**Issue by suarezluis** [wallabyjs/console-ninja#114]
# ⚠️ IMPORTANT 
If you have this problem, please update to Console Ninja `v1.0.105`+.
----

### What happened?

While using Console Ninja in a create-react-app eventually the browser errors with [ERR_NO_BUFFER_SPACE](https://stackoverflow.com/questions/75207113/err-no-buffer-space-error-on-m1-macbook-pro)

Removing the extension and restarting the IDE fixes the problem

### Version

v0.0.85

### Steps to reproduce (or sample repo)

The issue can't be reproduced constantly

By looking online seems to be affecting only apple sillycon devices 

### Log output

```shell
info host ------------ starting ------------
info extension host process output: port:[52048]
info host worker Starting Host Worker...
info extension 
info host client connected: extension { address: '::', family: 'IPv6', port: 57590 }
info host starting host: {"extensionFolder":"/Users/**********/.vscode/extensions/wallabyjs.console-ninja-0.0.85","runtimeHookFile":"/Users/**********/.vscode/extensions/wallabyjs.console-ninja-0.0.85/out/runtimeHook/tracer.js","globalErrorHandlerFile":"/Users/**********/.vscode/extensions/wallabyjs.console-ninja-0.0.85/out/runtimeHook/errorHandler.js","filesToInstrument":[{"path":"/Users/**********/Projects/**********/**********/healthcheck.js"},{"path":"/Users/**********/Projects/**********/**********/src/time.ts"},{"path":"/Users/**********/Projects/**********/**********/src/index.ts"},{"path":"/Users/**********/Projects/**********/**********/src/routes/twilio/handleSmsForward.ts"},{"path":"/Users/**********/Projects/**********/**********/src/routes/optOutHelpers.ts"},{"path":"/Users/**********/Projects/**********/**********/src/routes/twilio/handleTwilioTokenRequest.ts"},{"path":"/Users/**********/Projects/**********/**********/src/routes/bandwidth/handleHostingOrderStatus.ts"},{"path":"/Users/**********/Projects/**********/**********/src/services/loyaltyProgram/operationResults.ts"},{"path":"/Users/**********/Projects/**********/**********/src/devil/index.ts"},{"path":"/Users/**********/Projects/**********/**********/src/permissions/rules.ts"},{"path":"/Users/**********/Projects/**********/**********/src/resolvers/Twilio/twilio.resolvers.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/ChurnZero/churnZero.resolvers.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/CareSubscription/checkPetEnrollment.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/LoyaltyAccount/loyaltyAccount.resolvers.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/LoyaltyProgram/loyaltyProgram.resolvers.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/ClinicWidgetRequest/clinicWidgetRequest.resolvers.ts"},{"path":"/Users/**********/Projects/**********/**********/src/services/widget/createClinicWidgetRequest.ts"},{"path":"/Users/**********/Projects/**********/**********/src/services/widget/sendWidgetRequestToChannel.ts"},{"path":"/Users/**********/Projects/**********/**********/src/permissions/permissions.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/Reporting/reporting.resolvers.ts"},{"path":"/Users/**********/Projects/**********/**********/src/services/reports/index.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/ClinicPetParent/clinicPetParent.resolvers.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/ClinicPetParent/sendClinicPetParentAuthLink.ts"},{"path":"/Users/**********/Projects/**********/**********/src/services/careAuthentication/sendDirectedCareLoginLink.ts"},{"path":"/Users/**********/Projects/**********/**********/src/permissions/legalEntity.ts"},{"path":"/Users/**********/Projects/**********/**********/src/services/clinicRelations/updateClinicPetParentContactInfoFromPims.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/EmployerIdentificationNumber/employerIdentificationNumber.test.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/EmployerIdentificationNumber/EmployerIdentificationNumber.permissions.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/EmployerIdentificationNumber/EmployerIdentificationNumber.resolvers.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/EmployerIdentificationNumber/utils.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/ChannelMessage/channelMessage.resolvers.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/Search/search.resolvers.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/ChannelMessage/naturalLanguageTools.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/ActiveCampaign/emailCampaigns.resolvers.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/ChurnZero/churnZero.permissions.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/ChurnZero/churnZero.typeDefs.ts"},{"path":"/Users/**********/Projects/**********/**********/src/services/activeCampaign/index.ts"},{"path":"/Users/**********/Projects/**********/**********/src/services/activeCampaign/auth.ts"},{"path":"/Users/**********/Projects/**********/**********/src/services/activeCampaign/activeCampaignRequests.ts"},{"path":"/Users/**********/Projects/**********/**********/src/routes/webhooks.ts"},{"path":"/Users/**********/Projects/**********/**********/src/routes/twilio/index.ts"},{"path":"/Users/**********/Projects/**********/**********/src/routes/twilio/handleTwilioSmsForwardingHooks.ts"},{"path":"/Users/**********/Projects/**********/**********/src/routes/bandwidth/index.ts"},{"path":"/Users/**********/Projects/**********/**********/src/routes/bandwidth/handleInboundSms.ts"},{"path":"/Users/**********/Projects/**********/**********/src/routes/maybeSendSmsAutoResponse.ts"},{"path":"/Users/**********/Projects/**********/**********/src/prisma2/customResolvers/Channel/channel.resolvers.ts"},{"path":"/Users/**********/Projects/**********/**********/src/services/channels/channelExport.ts"},{"path":"/Users/**********/Projects/**********/**********/src/services/channels/createChannelMessage.ts"},{"path":"/Users/**********/Projects/**********/**********/pal.js"}],"debug":false,"config":{"dateTimeFormat":"hh:mm:ss.SSS","openLinksInEditor":true,"maxLogViewerEntries":15000,"fontSize":10,"autoClearLogViewerOnFileChanges":true,"outputMode":"Beside File","captureFunctions":false,"enableProFeatures":true,"toolsToAutoPatch":["vite","jest","webpack","next.js","cypress","http-server","serve","nuxt","remix","qwik","hydrogen"]},"allowedHosts":["localhost","127.0.0.1","example.cypress.io","Luiss-Mac-Studio.local","10.68.1.101"],"host":"127.0.0.1"}
info host client connected: extension { address: '::', family: 'IPv6', port: 52048 }
info host starting tools discovery and integration
info host looking for tools to patch
info host starting host: {"extensionFolder":"/Users/**********/.vscode/extensions/wallabyjs.console-ninja-0.0.85","runtimeHookFile":"/Users/**********/.vscode/extensions/wallabyjs.console-ninja-0.0.85/out/runtimeHook/tracer.js","globalErrorHandlerFile":"/Users/**********/.vscode/extensions/wallabyjs.console-ninja-0.0.85/out/runtimeHook/errorHandler.js","filesToInstrument":[{"path":"/Users/**********/Projects/**********/**********/src/pages/Application/CaptureScreenshot.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/pages/Application/index.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/pages/SuccessCenter/SuccessCenter.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/pages/Reporting/LookerEmbeddedReport.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/pages/AutomationsLegacy/useWorkflowDraft/saveToDb.ts"},{"path":"/Users/**********/Projects/**********/**********/src/pages/AutomationsLegacy/useWorkflowDraft/reducer.ts"},{"path":"/Users/**********/Projects/**********/**********/src/pages/Board/index.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/pages/Conversations/ChannelList/components/ChannelListFilterEdit.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/pages/Conversations/components/CustomizeExportedConversationNameModal.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/pages/AutomationsLegacy/useWorkflowDraft/index.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/pages/AutomationsLegacy/WorkflowSteps/Shared/ButtonActions.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/pages/Conversations/components/InMessageButtons/index.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/pages/Conversations/components/MessageBubbleAttachment/FormExport/FormExportModal.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/pages/AutomationsLegacy/WorkflowDrafting/WorkflowTriggering/TriggerTiming.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/shared/hooks/useTagManager.ts"},{"path":"/Users/**********/Projects/**********/**********/src/shared/hooks/useHotjar.ts"},{"path":"/Users/**********/Projects/**********/**********/src/shared/hooks/useLogRocket.ts"},{"path":"/Users/**********/Projects/**********/**********/src/pages/Settings/components/VerifyPhoneNumberModal.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/pages/Settings/pages/GeneralInfoSettings/EmployerIdentificationNumberModal.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/pages/Forms/FormsContent/FormsDrafting/Prompt.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/pages/Settings/pages/AppointmentSettings/index.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/shared/hooks/useChurnZero.ts"},{"path":"/Users/**********/Projects/**********/**********/src/shared/utils/mixpanel/useMixpanelTracking.ts"},{"path":"/Users/**********/Projects/**********/**********/src/pages/Settings/pages/TextMessagingSettings/EnablePhoneNumber.tsx"},{"path":"/Users/**********/Projects/**********/**********/graphql.config.js"},{"path":"/Users/**********/Projects/**********/**********/src/shared/hooks/useMixpanel.ts"},{"path":"/Users/**********/Projects/**********/**********/src/serviceWorker.js"},{"path":"/Users/**********/Projects/**********/**********/src/shared/hooks/useAutoPageReload.ts"},{"path":"/Users/**********/Projects/**********/**********/src/shared/components/SidePanel/components/MassTextAlerts/MassTextAlertsSendConfirmationModal.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/shared/components/SidePanel/components/MassTextAlerts/MassTextAlertsDateRangeSelector.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/shared/utils/devMode.ts"},{"path":"/Users/**********/Projects/**********/**********/src/shared/utils/mixpanel/mixpanel.ts"},{"path":"/Users/**********/Projects/**********/**********/src/shared/components/UnsupportedBrowserPopover/checkBrowserSupport.ts"},{"path":"/Users/**********/Projects/**********/**********/src/shared/components/PhoneCallPopover/index.tsx"},{"path":"/Users/**********/Projects/**********/**********/src/shared/providers/UnreadMessageCountProvider/index.tsx"},{"path":"/Users/**********/Projects/**********/**********/lint.js"},{"path":"/Users/**********/Projects/**********/**********/codegen.js"},{"path":"/Users/**********/Projects/**********/**********/clean-graphql.ts.js"},{"path":"/Users/**********/Projects/**********/**********/public/index.html"},{"path":"/Users/**********/Projects/**********/**********/src/routes/index.tsx"}],"debug":false,"config":{"dateTimeFormat":"hh:mm:ss.SSS","openLinksInEditor":true,"maxLogViewerEntries":15000,"fontSize":10,"autoClearLogViewerOnFileChanges":true,"outputMode":"Beside File","captureFunctions":false,"enableProFeatures":true,"toolsToAutoPatch":["vite","jest","webpack","next.js","cypress","http-server","serve","nuxt","remix","qwik","hydrogen"]},"allowedHosts":["localhost","127.0.0.1","example.cypress.io","Luiss-Mac-Studio.local","10.68.1.101"],"host":"127.0.0.1"}
info host starting tools discovery and integration
info host looking for tools to patch
info host client connected: runtimeHook { address: '::', family: 'IPv6', port: 52048 }
info host found 2 tool file(s) to patch
info host patching /Users/**********/Projects/**********/**********/node_modules/jest/bin/jest.js
info host patching /Users/**********/Projects/**********/**********/node_modules/jest-runner/build/testWorker.js
info host completed tools discovery and integration
info host found 3 tool file(s) to patch
info host patching /Users/**********/Projects/**********/**********/node_modules/webpack/lib/webpack.js
info host patching /Users/**********/Projects/**********/**********/node_modules/jest/bin/jest.js
info host patching /Users/**********/Projects/**********/**********/node_modules/jest-runner/build/testWorker.js
info host completed tools discovery and integration
info buildHook-70848 allowed tools: vite, jest, webpack, next.js, cypress, http-server, serve, nuxt, remix, qwik, hydrogen
info buildHook-70848 running tools: webpack
info host client connected: buildHook { address: '::', family: 'IPv6', port: 52048 }
info buildHook-70848 allowed tools: vite, jest, webpack, next.js, cypress, http-server, serve, nuxt, remix, qwik, hydrogen
info buildHook-70848 running tools: webpack
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 2 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 6 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 2 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 4 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: true
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 13 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 4 log points, error handler: false
info buildHook-70848 file processed (stat): 3 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 7 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 2 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: false
info buildHook-70848 file processed (stat): 0 log points, error handler: false
info buildHook-70848 file processed (sync): 0 log points, error handler: false
info buildHook-70848 file processed (stat): 1 log points, error handler: true
info host client disconnected: runtimeHook 1001
info host client disconnected: runtimeHook
info host client connected: runtimeHook { address: '::', family: 'IPv6', port: 52048 }
info host client connected: runtimeHook { address: '::', family: 'IPv6', port: 52048 }
info extension stopping host
info buildHook-70848 client disconnected: 1006 
info buildHook-70848 disconnected from host: 52048
info buildHook-70848 disconnected from all hosts
info host removing cache folder: d0dab48d0c7f84609db9da9bea7fcf1f
info extension stopping host
info host removing cache folder: 35a40475f888dfb36bc8d20db86a7d25
info buildHook-23565 installing build hook for webpack at /Users/**********/Projects/**********/**********
info buildHook-24775 installing build hook for webpack at /Users/**********/Projects/**********/**********
```

**Comment by ArtemGovorov** [comment]
Thanks for reporting the issue. 

When it happens, does browser page refresh help?

**Comment by sawyerclick** [comment]
I frequently run into this issue on various M1 Macs. Today, I received this error while attempting to git push for the first time. My entire computer froze, lost internet connection, and had to totally restart. I've narrowed it down to this extension as the culprit by disabling each I have installed one-by-one.

Will have to uninstall until this is addressed

**Comment by ArtemGovorov** [comment]
We are investigating the issue, and while we haven't been able to locally reproduced it, we have a few ideas to try and address it. We'll be releasing a new version with the fix attempt early next week.

The issue seems to be related to the volume of websocket writes on M1. Console Ninja only creates few websocket connections, but may send a lot of data over it over time (depending on how much `console.log` is used and logs in user code).

The M1 issue seems to be affecting other tools as well, even more severely it seems: https://github.com/lensapp/lens/issues/6063.

@suarezluis @sawyerclick, and **everyone else watching/encountering the issue**, to help us address it faster, can you please share some details about your use of the extension? 
- did you just have it installed and didn't use much, or you had it running for your projects and had a few `console.log` calls in your code? If so, how much data was logged - just few objects/strings, or a lot of data was logged?
- do you have autosave setting on in your VS Code for your file changes, or manually saving files when required?
- do you usually have 1-3 VS Code projects you're working with, or you typically have 5+ opened projects (with Console Ninja running) and multiple browser pages (with hot reload from your build tool) opened at the same time?

**Comment by ArtemGovorov** [comment]
We have released some changes that may fix the issue in v0.0.87+. 

@suarezluis @sawyerclick Can you please try the latest version and let us know if the issue is fixed for you?
If the issue is still there, please share some more details:
- did you just have it installed and didn't use much, or you had it running for your projects and had a few console.log calls in your code? If so, how much data was logged - just few objects/strings, or a lot of data was logged?
- do you have autosave setting on in your VS Code for your file changes, or manually saving files when required?
- do you usually have 1-3 VS Code projects you're working with, or you typically have 5+ opened projects (with Console Ninja running) and multiple browser pages (with hot reload from your build tool) opened at the same time?
- do you have any active VPN client or virus scanner (or any other tool that may be affecting/monitoring your network in any way)?

**Comment by suarezluis** [comment]
@ArtemGovorov Thanks for the update, will try it with the new version and get back to you

**Comment by eric-helix-93** [comment]
I'm still getting the issue on version v0.0.87.

> did you just have it installed and didn't use much, or you had it running for your projects and had a few console.log calls in your code? If so, how much data was logged - just few objects/strings, or a lot of data was logged?

I just installed this yesterday and only used it for about 2 hours. It was only logging strings and errors.

> do you have autosave setting on in your VS Code for your file changes, or manually saving files when required?

No autosave, manually save the files

> do you usually have 1-3 VS Code projects you're working with, or you typically have 5+ opened projects (with Console Ninja running) and multiple browser pages (with hot reload from your build tool) opened at the same time?

Only had the one VS code project open and running at the time and only one tab with the project running.

> do you have any active VPN client or virus scanner (or any other tool that may be affecting/monitoring your network in any way)?

No active VPN client or virus scanner

**Comment by smcenlly** [comment]
One more question for you @eric-helix-93 - can you tell us a little about your project type and logging behavior?

* What project type were you using? E.g. `create-react-app`? `vite`? `vite` with `react`? etc?
* Were you logging lots of values or only a small amount? I'd imagine the strings were fairly small? Did you get a lot of errors?

Also wanted to check that you're using Mac M1?

**Comment by eric-helix-93** [comment]
@smcenlly 

> What project type were you using? E.g. create-react-app? vite? vite with react? etc?

It's a vite with react app.

> Were you logging lots of values or only a small amount? I'd imagine the strings were fairly small? Did you get a lot of errors?

They were fairly small strings but I was getting a lot of errors as I was testing some API changes. We are using react query and a few of the logs and errors were triggering from the responses received from the API.

I am using an M1 Mac :)

**Comment by suarezluis** [comment]
Same here, issue persists, running single project on VsCode, create react app, M1 Ultra processor.

**Comment by ArtemGovorov** [comment]
@suarezluis Were you logging lots of values or only a small amount? Did you get a lot of errors logged?

**Comment by dilincoln** [comment]
Same here, macOS Ventura

**Comment by creativeindustriesgroup** [comment]
Same issue here - Pre-M1 Chip.

**Device & OS:**
MacBook Pro 2019 15"
2.4 GHz 8-Core Intel Core i9
Radeon Pro Vega 20 4 GB
Intel UHD Graphics 630 1536 MB
32 GB 2400 MHz DDR4
13.2.1 (22D68) (Also tested before the most recent update with the same issue)
Running 3x Displays

**Environment:**
Visual Studio Code: 
Version: 1.76.2 (Universal)
Date: 2023-03-14T17:54:09.061Z (1 wk ago)
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Darwin x64 22.3.0
Auto Save Enabled

**VSC Extensions:** 
GitLens
Import cost
PHP Server
Prettier
Simple React Snippets
VS Code Counter
Console Ninja v0.0.91

I seem to be able to run with no issues for around 2 hours and then will get the NO_BUFFER_SPACE error affecting all internet connection on the device. Shortly after, my Mac also Kernel Panics. I have deactivated Console Ninja and will report back if I can confirm that this is the source of my Kernel Panic/NO_BUFFER_SPACE error.

I have no VPN but do run Malwarebytes in the background. Browser Refresh, Restart does not work. Restarting `react-scripts start` does not work.

I access the site via http://localhost:3000 rather than via the LAN IP.

Hope this helps.

**Comment by smcenlly** [comment]
Can you please:

1. Provide us with your `Diagnostics reports` and `Crash Reports` ([instructions here](https://www.lifewire.com/troubleshoot-mac-os-x-kernel-panics-2260849#toc-kernel-panic-logs))
2. If possible, provide a sample application that has the same problem (we would like to understand dependencies and project setup / configuration).

**Comment by smcenlly** [comment]
@eric-helix-93, @creativeindustriesgroup, @suarezluis - can you please provide us with your MacOS `Diagnostics reports` and `Crash Reports` ([instructions here] so we can investigate further?

**Comment by creativeindustriesgroup** [comment]
> @eric-helix-93, @creativeindustriesgroup, @suarezluis - can you please provide us with your MacOS `Diagnostics reports` and `Crash Reports` ([instructions here] so we can investigate further?

Unfortunately I can't narrow down a specific time that the last panic happened. As soon as it happens again I'll get the time and send the logs over, unfortunately as I don't have a specific way of re-creating its a bit of a waiting game. 

Some more environment factors which may or may not help: 
From package.json
```
"private": true,
  "dependencies": {
    "@airbrake/browser": "^2.1.8",
    "@fortawesome/fontawesome-svg-core": "^6.2.1",
    "@fortawesome/free-brands-svg-icons": "^6.2.1",
    "@fortawesome/free-solid-svg-icons": "^6.2.1",
    "@fortawesome/react-fontawesome": "^0.2.0",
    "@stripe/react-stripe-js": "^1.16.4",
    "@stripe/stripe-js": "^1.46.0",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^13.5.0",
    "bulma": "^0.9.4",
    "clsx": "^1.2.1",
    "crisp-sdk-web": "^1.0.12",
    "date-fns": "^2.29.3",
    "dotenv": "^16.0.3",
    "firebase": "^9.15.0",
    "logrocket": "^3.0.1",
    "react": "^18.2.0",
    "react-bulma-components": "^4.1.0",
    "react-content-loader": "^6.2.0",
    "react-dom": "^18.2.0",
    "react-router-dom": "^6.6.2",
    "react-toastify": "^9.1.1",
    "sass": "^1.57.1",
    "web-vitals": "^2.1.4"
  },
  "devDependencies": {
    "react-scripts": "5.0.1"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
```

Some logs which I cannot confirm are from the correct crash/kernel panic but are around a similar time relate to the `remoted` process timing out/not responding, Adobe Creative Cloud `CCXProcess` and the `watchman` process. Again this is only suggestive for now until I can accurately pick out a crash report and even though Adobe CC is listed the crashing/network issues did go away when I uninstalled Console Ninja.

**Comment by smcenlly** [comment]
Yesterday we were able to reproduce the issue for ourselves. We were able to recover from the problem by killing all the associated console ninja processes:

```shell
ps -A | grep wallabyjs.console-ninja- | awk '{print $1}' | xargs kill
```

Please note we are working on this issue as a high priority.

**Comment by smcenlly** [comment]
Today we identified and fixed the problem that was causing this issue.

The problem was actually a bug/behavior in one of our dependencies, `node-fetch` and they way that we were using it. To optimize performance, we were returning content immediately without processing the response body in some cases (the requirement to process the body is not explicitly mentioned in their docs). This is a known bug that can lead to sockets/file descriptors being left open and is apparently an issue on MacOS and Linux (see https://github.com/node-fetch/node-fetch/issues/1673).

The issue would exhibit when multiple errors occurred when running in a browser environment when we were using `node-fetch` to retrieve source maps for error processing

The problem is fixed in Console Ninja `v1.0.105`.
